### ISSUE TITLE
vdpau's waf configure test fails if no vdpau devices exist at build time

### ISSUE THREAD

**Issue by codyopel** [mpv-player/mpv#3766]
### mpv version and platform
current revision
https://github.com/mpv-player/mpv/blob/3f5b41dfa30ca282fd99176bf879493dd72b3119/wscript#L899#L901

### Reproduction steps
Build in a chroot with `--enable-vdpau-hwaccel` and the configure test will fail.

### Actual behavior
The current test uses  `av_vdpau_bind_context(0,0,0,AV_HWACCEL_FLAG_ALLOW_HIGH_DEPTH)` from [libavcodec/vdpau.h](https://github.com/FFmpeg/FFmpeg/blob/415f907ce8dcca87c9e7cfdc954b92df399d3d80/libavcodec/vdpau.h#L169#L170) which requires a device to be passed.  When building on a system or chroot with no device the configure test will fail.

Currently I added a hack that replaces the configure test with the libdl test to get it to build.
```
      sed -i wscript \
        -e 's,libavcodec/vdpau.h,dlfcn.h,' \
        -e "s/.*av_vdpau_bind_context.*/'dlopen\(\"\", 0)',/"
```
Then built mpv and tested with a know working file and everything worked as expected.
`Using hardware decoding (vdpau).`

**Comment by mc4man** [comment]
Here the diff is whether using a shared FFmpeg vs. a static one. Plus there are 3 other config items that also succeed on shared, fail on static.
Brought up here - #3664
Don't see the point of replacing a failing check statement with some unrelated one

**Comment by codyopel** [comment]
@mc4man 
The sed hack is a simple quick and dirty test that will force waf to return true, ensuring that vdpau is enabled.

The failure has nothing to do with whether the library was statically or dynamically linked.  The configure test calls a function that that requires a vdpau device to exists, which may not be the case at build time.

**Comment by codyopel** [comment]
resolved by: f59371de2170141fc28540d03c4e7ecc62844ebf
